src/api/blob/controller.go

```diff
@@ -0,0 +1,238 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/distribution"
+	"github.com/garyburd/redigo/redis"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	util "github.com/goharbor/harbor/src/common/utils/redis"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/blob"
+)
+
+var (
+	// Ctl is a global blob controller instance
+	Ctl = NewController()
+)
+
+// Controller defines the operations related with blobs
+type Controller interface {
```

Comment:
basiclly, this controller operates three DB tables with paramemters, please add details comments for each method, otherwise it's hard to maintain.